mobile_index/src/settings.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{Error, Result};
+use config::{Config, Environment, File};
+use http::Uri;
+use serde::Deserialize;
+use sqlx::{postgres::PgPoolOptions, Pool, Postgres};
+use std::{path::Path, time};
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    /// RUST_LOG compatible settings string
+    pub log: Option<String>,
+    /// Check interval in seconds. (Default is 900; 15 minutes)
+    #[serde(default = "default_interval")]
+    pub interval: u64,
+    pub database: Database,
+    pub verifier: file_store::FileStoreSettings,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Database {
+    /// Max open connections to the database. (Default 10)
+    #[serde(default = "default_db_connections")]
+    pub max_connections: u32,
+    /// URL to access the postgres database. For example:
+    /// =postgres://postgres:postgres@127.0.0.1:5432/mobile_index_db
+    #[serde(with = "http_serde::uri")]
+    pub url: Uri,
+}
+
+impl Settings {
+    /// Load Settings from a given path. Settings are loaded from a given
+    /// optional path and can be overriden with environment variables.
+    ///
```

Comment:
Config is expected in TOML? Where are the valid values enumerated? Is there a default / expected path for config files?

Reply:
Each attribute name in the Settings struct is either a value (like interval) or a section if it's a set of values.

Reply:
@jadeallenx I added settings-template.toml example file with inline documentation